datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/join/JoinType.java

```diff
@@ -0,0 +1,229 @@
+
+
+
+  @Nonnull
+  @Override
+  public List<BrowsePath> browsePaths(@Nonnull String urn, @Nonnull QueryContext context) throws Exception {
+    final StringArray result = _entityClient.getBrowsePaths(getJoinUrn(urn), context.getAuthentication());
+    return BrowsePathsMapper.map(result);
+  }
+
+  private com.linkedin.common.urn.JoinUrn getJoinUrn(String urnStr) {
+    try {
+      return JoinUrn.createFromString(urnStr);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(String.format("Failed to retrieve data product with urn %s, invalid urn", urnStr));
+    }
+  }
+
+  @Override
+  public SearchResults search(@Nonnull String query, @Nullable List<FacetFilterInput> filters,
+      int start, int count, @Nonnull QueryContext context) throws Exception {
+    final Map<String, String> facetFilters = ResolverUtils.buildFacetFilters(filters, FACET_FIELDS);
+    final SearchResult searchResult = _entityClient.search(ENTITY_NAME, query, facetFilters, start,
+        count, context.getAuthentication(), new SearchFlags().setFulltext(true));
+    return UrnSearchResultsMapper.map(searchResult);
+
+  }
+
+  @Override
+  public AutoCompleteResults autoComplete(@Nonnull String query, @Nullable String field,
+      @Nullable Filter filters, int limit, @Nonnull QueryContext context) throws Exception {
+    final AutoCompleteResult result = _entityClient.autoComplete(ENTITY_NAME, query, filters, limit, context.getAuthentication());
+    return AutoCompleteResultsMapper.map(result);
+  }
+
+  @Override
+  public Join update(String urn, @Nonnull JoinUpdateInput input, @Nonnull QueryContext context)
+      throws Exception {
+    if (isAuthorized(urn, input, context)) {
+      final CorpuserUrn actor = CorpuserUrn.createFromString(context.getAuthentication().getActor().toUrnStr());
+
+      // Same routine used by create - hence this check
+      JoinUrn inputUrn = new JoinUrn(UUID.randomUUID().toString());
+      if (urn != null) {
+        inputUrn = JoinUrn.createFromString(urn);
+        if ("new".equals(inputUrn.getJoinIdEntity())) {
+          inputUrn = new JoinUrn(UUID.randomUUID().toString());
+        }
+      } else {
+        urn = inputUrn.toString();
+      }
+
+      final JoinUrn updatedUrn = inputUrn;
+
+      final Collection<MetadataChangeProposal> proposals = JoinUpdateInputMapper.map(input, actor);
+      proposals.forEach(proposal -> proposal.setEntityUrn(updatedUrn));
+
+      try {
+        _entityClient.batchIngestProposals(proposals, context.getAuthentication(), false);
+      } catch (RemoteInvocationException e) {
+        throw new RuntimeException(String.format("Failed to write entity with urn %s", urn), e);
+      }
+
+      return load(urn, context).getData();
+    }
+    throw new AuthorizationException("Unauthorized to perform this action. Please contact your DataHub administrator.");
+  }
+
+  private boolean isAuthorized(String urn, JoinUpdateInput input, QueryContext context) {
+    return true;
+  }
```

Comment:
Thank you @chriscollins3456 for your inputs here!

> we're going to want some permissions around managing joins on datasets. it would have to check to make sure the user has permission to manage a dataset's joins. so i'd suggest an `EDIT_JOINS_PRIVILEGE` like we have `EDIT_QUERIES_PRIVILEGE` - we can just have one permission now that we check for all CRUD actions

Regarding this we had a discussion with our product team and can implement as below:
1. We can have a policy on CreateJoin on Datasets entity and individual installations can setup as per requirements. (Ex. If they only want dataset owners to create join)
2. As Join is an Entity unlike Queries over Dataset it also has its owner list so we can manage Join's editing using EDIT_ENTITY privilege on Join Entity to allow only join entity's owners to edit Joins.

Please provide your suggestions for the same.

Reply:
Addressed